### PR TITLE
Add Tekton orchestrator

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes

This PR adds an alpha-version of a potential Tekton orchestrator in ZenML.

**Missing**:
- Docs
- Example
- Tests

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/mlops-stacks/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/tekton/orchestrators/__init__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
```
Comment (htahir1) [inline src/zenml/integrations/tekton/orchestrators/__init__.py:1]: 2022

**Review by htahir1 — APPROVED** [review]
Looks great!

**Comment by htahir1** [comment]
Added the example and docs

**Review by fa9r — APPROVED** [review]
Awesome, looks good!

The only suggestion I'd have is to refactor some of the duplicate code from KFP orchestrator, but I guess we can do that at a later point as part of ENG-982.

Otherwise looks very polished, ready to be merged 👍

**Comment by htahir1** [comment]
@strickvl if you give an approval we can merge!

File: docs/book/mlops-stacks/orchestrators/tekton.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+To use the Tekton orchestrator, we need:
+* The ZenML `tekton` integration installed. If you haven't done so, run 
+    ```shell
+    zenml integration install tekton -y
+    ```
+* [Docker](https://www.docker.com) installed and running.
+* [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) installed.
+
+When using the Tekton orchestrator with a remote cluster, you'll additionally need
```
Comment (schustmi) [inline docs/book/mlops-stacks/orchestrators/tekton.md:98]: There is no `local` case at the moment, I think we can remove this sentence right?

File: examples/tekton_pipeline_orchestration/setup.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
```
Comment (schustmi) [inline examples/tekton_pipeline_orchestration/setup.sh:1]: I think the `setup.sh` file should only exist for examples that can be run locally, so should be removed in this case

Comment (htahir1) [reply examples/tekton_pipeline_orchestration/setup.sh:1]: Ok! Shouldn't we also remove this from the kubeflow pipelines example too?

File: docs/book/mlops-stacks/orchestrators/tekton.md (hunk 2 above)
Comment (htahir1) [reply docs/book/mlops-stacks/orchestrators/tekton.md:98]: Hmm but the local case isn't mentioned here. Do you mean saying the word "additionally"?

File: examples/tekton_pipeline_orchestration/setup.sh (hunk 3 above)
Comment (schustmi) [reply examples/tekton_pipeline_orchestration/setup.sh:1]: I think the kubeflow example actually runs locally (assuming that you have the pre-requisites installed), but Tekton doesn't support that (for now)

File: docs/book/mlops-stacks/orchestrators/tekton.md (hunk 2 above)
Comment (schustmi) [reply docs/book/mlops-stacks/orchestrators/tekton.md:98]: Yeah, I think we could just merge these two lists and remove the intermediate sentence, these are all equivalent requirements to run it.

**Pull request merged**